### PR TITLE
refactor(api): change auth to be cookie-based

### PR DESCRIPTION
**Pull request by kyranet**
Pending for skyra.pw to be updated with the new code.

### CODE REVIEW

**Review by Quantumlyy — DISMISSED** [review]

**Comment by favna** [comment]
Dashboard is ready, master has been updated as of https://github.com/skyra-project/skyra.pw/commit/03cd4a64925e953569d9c2dbdb89d4c9ad2a06eb.

File: src/middlewares/authenticated.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { ApiRequest, UserAuthObject } from '@lib/structures/api/ApiRequest';
+import { ApiResponse } from '@lib/structures/api/ApiResponse';
+import { Middleware, MiddlewareStore, Route, Util } from 'klasa-dashboard-hooks';
+
+export default class extends Middleware {
+
+	public constructor(store: MiddlewareStore, file: string[], directory: string) {
```
Comment (favna) [inline src/middlewares/authenticated.ts]: Middleware extends piece doesn't it? If so ApplyOptions please

File: src/middlewares/cookies.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { Middleware, MiddlewareStore } from 'klasa-dashboard-hooks';
+import { ApiRequest } from '@lib/structures/api/ApiRequest';
+import { ApiResponse } from '@lib/structures/api/ApiResponse';
+import { CookieStore } from '@lib/structures/api/CookieStore';
+import { DEV } from '@root/config';
+
+export default class extends Middleware {
+
+	public constructor(store: MiddlewareStore, file: string[], directory: string) {
+		super(store, file, directory, { priority: 30 });
```
Comment (favna) [inline src/middlewares/cookies.ts]: Ditto

**Review by favna — CHANGES_REQUESTED** [review]

**Comment by favna** [comment]
There's some todo's left for the dashboard

- logout should call logout endpoint rather than only pruning locally

- checking that 403's in sync user go to logout

**Review by favna — APPROVED** [review]

**Review by Quantumlyy — APPROVED** [review]

**Pull request merged**